src/components/WeatherIcon.tsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { View, Image, StyleSheet } from 'react-native';
+
+export enum WeatherType {
+    Sunny,
+    Thunderstorm,
+}
+
+interface IImageData {
+    src: number;
+    w: number;
+    h: number;
+}
+
+type WeatherToImage = {
+    [key in WeatherType]: IImageData;
+};
+
+const Images: WeatherToImage = {
+    [WeatherType.Sunny]: {
+        src: require('@assets/icons/sun.png'),
+        w: 35,
+        h: 35,
+    },
+    [WeatherType.Thunderstorm]: {
+        src: require('@assets/icons/thunderstorm.png'),
+        w: 42,
+        h: 33,
+    },
+};
+
+export const WeatherIcon: React.FC<{ weatherType: WeatherType }> = props => {
+    const image = Images[props.weatherType];
+    return (
+        <View style={styles.wrapper}>
+            <Image source={image.src} style={{ width: image.w, height: image.h }} />
```

Comment:
You can add `resizeMode: 'contain'` to the styles of Image, so its image would resize automatically to fit view, also it would help if the actual size of png image is bigger than what we need. When using it we don't need to specify height of image at all